src/Compilers/Core/Portable/FileSystem/FileUtilities.cs

```diff
@@ -398,29 +398,5 @@ internal static long GetFileLength(string fullPath)
                 throw new IOException(e.Message, e);
             }
         }
-
-        internal static Stream OpenFileStream(string path)
```

Comment:
>OpenFileStream [](start = 31, length = 14)

Why not keep it here and call it from StandardFileSystem?

Reply:
The problem here is the exception wrapping / normalization. When mocking out the file system you have to decide for a given call what exception should I throw? If you are just trying to replicate a file system failure the easiest path is to just throw the exception the file system would throw. Then have all the code on top actually do the wrapping. 

If we keep this method and have `StandardFileSystem` call into it then we can't take that simple approach. Instead we have to change the exception thrown from `IFileSystem` to match the wrapping semantics of the callsite which invoked the member. That's just asking for bugs and flaky behavior.

Reply:
I don't see how it makes easier to mock if you need to test all the possible exceptions a file system may throw vs. 3 exceptions that OpenFileStream may throw: ArgumentException, FileNotFoundException and IOException.

---
In reply to: [603561499](https://github.com/dotnet/roslyn/pull/52208#discussion_r603561499) [](ancestors = 603561499)

Reply:
The contract of IFileSystem.OpenFileWithNormalizedException is that only these 3 exceptions are thrown, the contract of IFileSystem.OpenFile is that any exception can be thrown that might be thrown by the underlying FS. If the contract of IFileSystem.OpenFile was the former it would imo be avoid bugs since the caller wouldn't need to decide whether to call OpenFileWithNormalizedException or OpenFile and wrap exceptions.

---
In reply to: [603570561](https://github.com/dotnet/roslyn/pull/52208#discussion_r603570561) [](ancestors = 603570561,603561499)

Reply:
Let's consider the proposed change here which is to essentially keep this method and have `StandardFileSystem.OpenStream` forward directly to this. That means there is no correct way to implement `IFileSystem.OpenStream` because the behavior should change depending on *where* the method is called from. That's because you've tied the implementation of `IFileSystem` to the wrapping behavior that we implement but do not universally consume in roslyn. 

Yes I agree most of our code should use a standard wrapping function for simplicity but at the moment we do not. Hence in order to faithfully replicate how the file system works we have to break apart the concepts in the testing layer. Otherwise we can't adequately represent what the file system can do today.

Reply:
Ok, I see your point. Shouldn't we follow up then with a change (or an issue) that aims to use standard wrapping across the compiler?

Reply:
Filed this to track making it consistent https://github.com/dotnet/roslyn/issues/52241